rclrs/src/parameter.rs

```diff
@@ -605,9 +454,101 @@ enum ParameterStorage {
     Undeclared(ParameterValue),
 }
 
+impl ParameterStorage {
+    pub(crate) fn to_parameter_type(&self) -> u8 {
+        match self {
+            ParameterStorage::Declared(s) => match s.kind {
+                ParameterKind::Bool => ParameterType::PARAMETER_BOOL,
+                ParameterKind::Integer => ParameterType::PARAMETER_INTEGER,
+                ParameterKind::Double => ParameterType::PARAMETER_DOUBLE,
+                ParameterKind::String => ParameterType::PARAMETER_STRING,
+                ParameterKind::ByteArray => ParameterType::PARAMETER_BYTE_ARRAY,
+                ParameterKind::BoolArray => ParameterType::PARAMETER_BOOL_ARRAY,
+                ParameterKind::IntegerArray => ParameterType::PARAMETER_INTEGER_ARRAY,
+                ParameterKind::DoubleArray => ParameterType::PARAMETER_DOUBLE_ARRAY,
+                ParameterKind::StringArray => ParameterType::PARAMETER_STRING_ARRAY,
+                ParameterKind::Dynamic => match &s.value {
+                    DeclaredValue::Mandatory(v) => v.read().unwrap().rcl_parameter_type(),
+                    DeclaredValue::Optional(v) => v
+                        .read()
+                        .unwrap()
```

Comment:
I think documenting the `unwraps` is a reasonable idea, though this may mean that we need to dig through our already-written code to make sure any other `unwraps` are in compliance with these requirements.